### PR TITLE
ADEN-10864 | Unique adUnit and pos key-value for DistroScale

### PR DESCRIPTION
**Pull request by Tomasz-at-Fandom**

### CODE REVIEW

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts [hunk 1]
```diff
@@ -45,8 +47,38 @@ export class UcpSlotsStateSetup implements DiProcess {
 		}
 	}
 
+	private distroScaleIU(adSlot: AdSlot, params: VideoParams): string {
```
Comment (nandy-andy) [inline platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:50]: 2nd thought: I'd rename it to `getDistroscaleAdUnit()` if we decide to stay with it ;)

File: src/ad-engine/video/vast-url-builder.ts [hunk 2]
```diff
@@ -80,6 +80,7 @@ export function buildVastUrl(
 
 	if (slot) {
 		params.push(`iu=${slot.getVideoAdUnit()}`);
+
```
Comment (dmnsobczak) [inline src/ad-engine/video/vast-url-builder.ts:83]: Unnecessary new line

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts [hunk 3]
```diff
@@ -45,8 +47,38 @@ export class UcpSlotsStateSetup implements DiProcess {
...
+		const newAdSlot = new AdSlot({ id: slotName });
+		const distroscaleIU = this.distroScaleIU(newAdSlot, params);
+		window.ads.runtime = window.ads.runtime || ({} as Runtime);
+		window.ads.runtime.distroscale = window.ads.runtime.distroscale || {};
+		window.ads.runtime.distroscale.adUnit = distroscaleIU;
+	}
+
 	private setupIncontentPlayerForDistroScale(): void {
 		const slotName = 'incontent_player';
+		this.setDistroscaleVarInRuntime(slotName);
```
Comment (dmnsobczak) [inline platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:81]: I don't like `var` phrase xD Maybe `setupDistroscaleRuntime`?

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts [hunk 4]
```diff
@@ -45,8 +47,38 @@ export class UcpSlotsStateSetup implements DiProcess {
...
+	// TODO: make it more clean
+	private setDistroscaleVarInRuntime(slotName: string): void {
+		const params = {
+			slotName,
+			type: 'porvata3',
+			adProduct: 'incontent_veles',
+		};
+		const newAdSlot = new AdSlot({ id: slotName });
+		const distroscaleIU = this.distroScaleIU(newAdSlot, params);
+		window.ads.runtime = window.ads.runtime || ({} as Runtime);
```
Comment (dmnsobczak) [inline platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:74]: I would add new, blank line before this line, I don't know why.

**Review by dmnsobczak — APPROVED** [review]

**Comment by dmnsobczak** [comment]
Code is ok, few formatting suggestions left.

File: src/ad-products/common/product-info.ts [hunk 5]
```diff
@@ -33,3 +33,12 @@ export function getAdProductInfo(slotName, loadedTemplate, loadedProduct): AdPro
 		adProduct: product.toLowerCase(),
 	};
 }
+
+export function getAdUnitString(adslotName: string, slotConfig: {}): string {
```
Comment (preludek) [inline src/ad-products/common/product-info.ts:37]: shouldn't slotConfig be typed more specifically?  Both uses of this function uses group and adProduct keys.

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts [hunk 6]
```diff
@@ -45,8 +49,21 @@ export class UcpSlotsStateSetup implements DiProcess {
 		}
 	}
 
+	private setDistroscaleVarInRuntime(slotName: string): void {
+		const params = {
+			group: 'VIDEO',
+			adProduct: 'incontent_veles',
+			slotNameSuffix: '',
+		};
+
```
Comment (preludek) [inline platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:58]: I would delete this empty line 🙈

Comment (Tomasz-at-Fandom) [reply platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:58]: "They said I could be anything, so I became a linter" xD

Comment (Tomasz-at-Fandom) [reply platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:58]: No offense, I'm just suggesting to leave such details to the linter or prefix them with `nit:` to make it easier for the author to go through the comments :)

File: src/ad-engine/video/vast-url-builder.ts (hunk 2 above)
Comment (Tomasz-at-Fandom) [reply src/ad-engine/video/vast-url-builder.ts:83]: https://github.com/Wikia/ad-engine/pull/983/files/2124969ad2d100d1ba9c3f973ce2be7464ee093e#r634977474

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts (hunk 1 above)
Comment (Mariusz8Walczak) [reply platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts]: wywalone

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts (hunk 3 above)
Comment (Mariusz8Walczak) [reply platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts:66]: wywalone

File: platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts (hunk 4 above)
Comment (Mariusz8Walczak) [reply platforms/ucp/setup/state/slots/ucp-slots-state-setup.ts]: do starej wersji

**Review by preludek — APPROVED** [review]

**Pull request merged**